charts/endpoint-controller/templates/deployment.yaml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}
+  labels:
+    {{- include "endpoint-controller.labels" . | nindent 4}}
+spec:
+  replicas: {{.Values.replicaCount}}
+  selector:
+    matchLabels:
+      {{- include "endpoint-controller.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "endpoint-controller.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{include "endpoint-controller.serviceAccountName" .}}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{.Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion}}"
+          imagePullPolicy: {{.Values.image.pullPolicy}}
+          environment:
+              SYNC_PERIOD: {{.Values.controller.sync_period}}
```

Comment:
We have sane defaults in controller, I would rather allow to provide extra env vars from values.yaml

Reply:
for what purpose?
I think it's easier for end user to have it like this.